clusterdock/topologies/cdh/actions.py

```diff
@@ -185,13 +254,21 @@ def start(args):
                 "direct any feedback to our community forum at "
                 "http://tiny.cloudera.com/hadoop-101-forum.")
 
-def restart_cm_server(primary_node):
+def get_cloudera_nodes(cluster):
+    return cluster.get_nodes_by_node_group_names(['primary', 'secondary'])
+
+def create_users_on_cloudera_nodes(cluster, kerberos_principals):
+    commands = ['useradd -u {} -g hadoop {}'.format(uid, primary)
+                for uid, primary in enumerate(kerberos_principals.split(','), start=1000)]
```

Comment:
Might want to abstract the start uid as a constant somewhere and just add a comment saying why we make sure it's above 1000. (Guessing it's to make sure it's not blacklisted?)